src/main/java/org/terasology/launcher/game/GameStarter.java

```diff
@@ -77,4 +100,20 @@ public Process call() throws IOException {
     Path getRuntimePath() {
         return Paths.get(System.getProperty("java.home"), "bin", "java");
     }
+
+    String homeDirParameter(Path gameDataDirectory) {
+        if (terasologyUsesPosixOptions()) {
+            return "--homedir=" + gameDataDirectory.toAbsolutePath();
+        } else {
+            return "-homedir=" + gameDataDirectory.toAbsolutePath();
+        }
+    }
+
+    String noSplashParameter() {
+        return terasologyUsesPosixOptions() ? "--no-splash" : "-noSplash";
+    }
+
+    boolean terasologyUsesPosixOptions() {
+        return release.getId().getVersion().compareTo("5.1") >= 0;
```

Comment:
`String.compareTo` is _not_ a robust way to compare version identifiers.

and I'm not even sure if "5.1" is the type of version in that field, or if it's "alpha-20" or something else.

Reply:
Meh, this is probably one of the half-finished spots of my refactoring :see_no_evil: 

I'd like to switch this id to be a SemVer such that it is comparable, and `build` and `profile` define whether the game identifier is within the same realm.

Ideally, we could get rid of the `Build` enum because we can encode stable or nightly/preview releases via SemVer itself, which would only leave the `Profile` (Omega, bare engine, ...) as different realm.

For the actual content of that `version` string: For releases from github we try to parse the tag as semver and use that (so, yes, it's a semver):

https://github.com/MovingBlocks/TerasologyLauncher/blob/master/src/main/java/org/terasology/launcher/repositories/GithubRepositoryAdapter.java#L65

For builds from Jenkins we take the version from the `displayVersion` in the additional properties file, adding SemVer-compliant build information.

https://github.com/MovingBlocks/TerasologyLauncher/blob/master/src/main/java/org/terasology/launcher/repositories/JenkinsRepositoryAdapter.java#L97-L106

Unfortunately, this does not seem to be a valid semver :confused: Currently, the content of `versionInfo.properties` looks like this:

```properties
buildId=585
buildNumber=585
buildTag=jenkins-Terasology-engine-develop-585
buildUrl=http\://jenkins.terasology.io/teraorg/job/Terasology/job/engine/job/develop/585/
dateTime=2021-07-30T18\:50\:53Z
displayVersion=alpha-20
engineVersion=5.1.0-SNAPSHOT
gitCommit=5928a7ddd1430d86d3cf238253130d4e61f0d331
jobName=Terasology/engine/develop
```

This again boils down to the question whether we can find a consistent naming and versioning scheme for our releases....

Reply:
Yeah, I wouldn't expect anything named "display" to be a semver.

If you want the same type of thing that we name tags after for the MovingBlocks/Terasology repo, that's the `engineVersion` property there.

Reply:
Using #647, this is changed to Semver comparison on engineVersion.